src/MongoDB.Driver/Encryption/ClientEncryption.cs

```diff
@@ -0,0 +1,186 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Bson.IO;
+using MongoDB.Bson.Serialization.Serializers;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Libmongocrypt;
+
+namespace MongoDB.Driver
+{
+    /// <summary>
+    /// Explicit client encryption.
+    /// </summary>
+    public class ClientEncryption
+    {
+        // private fields
+        private readonly LibMongoCryptController _libMongoCryptController;
+
+        // constructors
+        internal ClientEncryption(LibMongoCryptController libMongoCryptController)
+        {
+            _libMongoCryptController = Ensure.IsNotNull(libMongoCryptController, nameof(libMongoCryptController));
+        }
+
+        // public methods
+        /// <summary>
+        /// Creates a data key.
+        /// </summary>
+        /// <param name="kmsProvider">The kms provider.</param>
+        /// <param name="dataKeyOptions">The data key options.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>A data key.</returns>
+        public BsonValue CreateDataKey(string kmsProvider, DataKeyOptions dataKeyOptions, CancellationToken cancellationToken)
+        {
+            var key = GetKmsProvider(kmsProvider, dataKeyOptions);
+            return new BsonBinaryData(_libMongoCryptController.GenerateKey(key, cancellationToken), GuidRepresentation.Standard);
```

Comment:
Note: one practical reason for not calling methods in the arguments of another method call is that it makes stepping in to a method with a debugger much easier. Otherwise you keep ending up in other methods first.